python/ray/serve/_private/application_state.py

```diff
@@ -1094,12 +1106,23 @@ def override_deployment_info(
         override_options["replica_config"] = replica_config
 
         # Override deployment config options
-        original_options = info.deployment_config.dict()
         options.pop("name", None)
         original_options.update(options)
         override_options["deployment_config"] = DeploymentConfig(**original_options)
         deployment_infos[deployment_name] = info.update(**override_options)
 
+        deployment_config = deployment_infos[deployment_name].deployment_config
+        if (
+            deployment_config.autoscaling_config is not None
+            and deployment_config.max_concurrent_queries
+            < deployment_config.autoscaling_config.target_num_ongoing_requests_per_replica  # noqa: E501
```

Comment:
Haha! The `#noqa` is a sign this param name is too long :)

Reply:
🤦‍♀️